src/reducers/user.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import { USER } from "../constants/actionTypes";
+import { type UserActions } from "../actions/user";
+import { type Profile } from "../types";
+
+export type State = {
+  auth: boolean,
+  error: string,
+  loading: boolean,
+  profile: ?Profile
+};
+const initialState: State = {
+  auth: false,
```

Comment:
auth?
Why do you need this value?
Anyway, logic for user and auth shouldn't be mixed.